tika-parsers/src/main/resources/org/apache/tika/parser/captioning/tf/im2txtapi.py

```diff
@@ -0,0 +1,220 @@
+    """
+        Given HTTP URL, this api gets the content of it
+        returns (Content-Type, image_content)
+    """
+    try:
+        app.logger.info("GET: %s" % url)
+        auth = None
+        res = requests.get(url, stream=True, timeout=timeout, auth=auth)
+        if res.status_code == success:
+            return res.headers.get('Content-Type', 'application/octet-stream'), res.raw.data
+    except:
+        pass
+    return None, None
+
+
+@app.route("/")
+def index():
+    """The index page which provide information about other API end points"""
+
+    return """
+    <div>
+    <h1> Image Captioning REST API </h1>
+    <h3> The following API end points are valid </h3>
+        <ul>
+            <h4> Inception V3 </h4>
+            <li> <code>/ping </code> - <br/>
+                <b> Description : </b> checks availability of the service. returns "pong" with status 200 when it is available
+            </li>
+            <li> <code>/getcaptions</code> - <br/>
+                <table>
+                <tr><th align="left"> Description </th><td> This is a service that can caption images</td></tr>
+                <tr><th align="left"> How to supply Image Content </th></tr>
+                <tr><th align="left"> With HTTP GET : </th> <td>
+                    Include a query parameter <code>url </code> which is an http url of JPEG image <br/>
+                    Example: <code> curl "localhost:8764/getcaptions?url=http://xyz.com/example.jpg"</code>
+                </td></tr>
+                <tr><th align="left"> With HTTP POST :</th><td>
+                    POST JPEG image content as binary data in request body. <br/>
+                    Example: <code> curl -X POST "localhost:8764/getcaptions" --data-binary @example.jpg </code>
+                </td></tr>
+                </table>
+            </li>
+        <ul>
+    </div>
+    """
+
+
+@app.route("/ping", methods=["GET"])
+def ping_pong():
+    """API to do health check. If this says status code 200, then healthy"""
+
+    return "pong"
+
+
+@app.route("/getcaptions", methods=["GET", "POST"])
+def get_captions():
+    """API to caption images"""
+
+    st = current_time()
+    # get beam_size
+    beam_size = int(request.args.get("beam_size", "3"))
+    # get max_caption_length
+    max_caption_length = int(request.args.get("max_caption_length", "20"))
+    # get image_data
+    if request.method == 'POST':
+        image_data = request.get_data()
+    else:
+        url = request.args.get("url")
+        c_type, image_data = get_remote_file(url)
+        if not image_data:
+            return flask.Response(status=400, response=jsonify(error="Could not HTTP GET %s" % url))
+        if 'image/jpeg' not in c_type:
+            return flask.Response(status=400, response=jsonify(error="Content of %s is not JPEG" % url))
+    read_time = current_time() - st
+
+    # restart counter
+    st = current_time()
+
+    generator = caption_generator.CaptionGenerator(app.model,
```

Comment:
Well, that's what I did initially. Than I measured time to make the CaptionGenerator object. **It was 0 ms**(It may be taking few nano seconds). I need to pass the post parameters beam_size & max_caption_length in order to make the CaptionGenerator object. If I put it in the initialization, client won't be able to change those parameters upon request, client will only be able to change them in initialization. I think giving users the ability to change the process of captioning an image in every request is nice rather than saving some nano seconds. What do you think?